KILTprotocol/mashnet-node
# refactor: migrate pallets' StorageVersion

## fixes KILTProtocol/ticket#1627
* Sets `StorageVersion` via pallet macro for all pallets (excluding Staking which is handled in #318)
* Adds Migrations for `DID` and `Delegation` pallets which already used a custom versioning scheme (as done in #318)

## How to test:

```
RUST_LOG=runtime=trace,try-runtime::cli=trace,executor=trace cargo run --release -p kilt-parachain --features try-runtime -- try-runtime --chain=spiritnet-dev --execution=Native --no-spec-name-check on-runtime-upgrade live --uri=wss://spiritnet.kilt.io:443/para  
```

## Checklist:

  - [x] No panics! (checked arithmetic ops, no indexing `array[3]` use `get(3)`, ...)
